### DISCUSSION TITLE
Mocking CDI beans in Quarkus

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#24909] [Quarkus Blog/Website]
# Mocking CDI beans in Quarkus

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/mocking/

**Comment by kepich** [comment]
You cannot do that:
    
    @InjectMock
    private InvoiceNotificationService invoiceNotificationService;

    @BeforeAll
    public static void setup() {
        doNothing().when(invoiceNotificationService).sendInvoice(any());
        doNothing().when(invoiceNotificationService).notifyInvoiceAlreadySent(any());
    }

You cannot use non static properties in static context

**Comment by mikib0** [comment]
> beans with @Singleton and @Dependent scope cannot be used with QuarkusMock.

any solution for mocking @Dependent beans? 

I know [there's for @Singleton beans](https://quarkus.io/guides/getting-started-testing#cdi-alternative-mechanism#:~:text=Mocking%20%40Singleton%20beans%20can%20be%20performed%20by%20adding%20the%20%40MockitoConfig(convertScopes%20%3D%20true)%20annotation.), but can't find mention of mocking @Dependent beans anywhere.

(Bonus question: why don't non normal scopes work with QuarkusMock/InjectMock?)

**Reply by mschorsch** [reply]
I'm not sure If this works but you can try in the static context `Arc.containter().instance(InvoiceNotificationService.class)`
